### DISCUSSION TITLE
Get Request IP and Port

### DISCUSSION THREAD

**Discussion by lonpalmer** [graphql-java/graphql-java#3341] [Q&A] [answered]
How do I get the requesting IP and Port of requests from within the DataFetcher?  I see from this release note that I can access either an GraphQLServletContext or GraphQLWebSocketContext but I have no idea how to get access to these interfaces.  Is it something I should be able to access from the DataFetchingEnvironment?

For context, I'm using Micronaut with GraphQL-Java.

**Comment by bbakerman** [comment, marked as answer]
This is a total guess since I have never programmed in Micronaut - but typically frameworks send in context data via the `graphql.GraphQLContext` object which is created inside the `graphql.ExecutionInput`

Each data fetcher gets access to this via `graphql.schema.DataFetchingEnvironment#getGraphQlContext`

Looking quickly at the Micronaught code it seems it has a  https://github.com/micronaut-projects/micronaut-graphql/blob/master/graphql/src/main/java/io/micronaut/configuration/graphql/GraphQLExecutionInputCustomizer.java that allows you to customize the `ExecutionInput` and it gives you the http request

```
public Publisher<ExecutionInput> customize(
            ExecutionInput executionInput,
            HttpRequest httpRequest,
            @Nullable MutableHttpResponse<String> httpResponse) {
        return Publishers.just(executionInput);
    }
```

I would look at ways to customize the ExecutionInput via this mechanism

**Reply by lonpalmer** [reply]
Thank you!
